app/src/ui/branches/PushBranchCommits.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Branch } from '../../models/branch'
+import { ButtonGroup } from '../../ui/lib/button-group'
+import { Button } from '../../ui/lib/button'
+import { Dialog, DialogContent, DialogFooter } from '../../ui/dialog'
+import { Repository } from '../../models/repository'
+
+interface IPushBranchCommitsProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly branch: Branch
+  readonly onConfirm: (repository: Repository, branch: Branch) => void
+  readonly onDismissed: () => void
+
+  /**
+   * When set to true, the publish branch dialog is returned.
+   *
+   * @type {boolean}
+   * @memberof IPushBranchCommitsProps
+   */
+  readonly publish?: boolean
```

Comment:
It seems like this and `unPushedCommits` are communicating the same information, so we should be able to infer this from whether `unPushedCommits` is defined/greater than zero.